### PR TITLE
Dek collect targeted pcr metrics it3

### PR DESCRIPTION
**Pull request by dekling**
Created new branch (old one was problematic).  Integration test was successful.  Let me know if additional parameters should be tested.

### CODE REVIEW

File: src/test/java/org/broadinstitute/hellbender/tools/picard/analysis/directed/CollectTargetedPcrMetricsIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    @Test
+    public void testCollect () throws IOException {
+
+        final File input = new File(TEST_DATA_PATH, "first5Kamod_reordered.bam");
+        final File amplicon_intervals = new File(TEST_DATA_PATH, "intervals_b37_20_1.interval_list");
+        final File target_intervals = new File(TEST_DATA_PATH, "intervals_b37_20_1.interval_list");
+        final File expectedFile = new File(TEST_DATA_PATH, "PCRMetrics_new_bam.txt");
+        final File outfile = BaseTest.createTempFile("PCRMetrics", ".txt");
+        final File reference = new File(largeFileTestDir, "human_g1k_v37.20.21.fasta");
+        final File pertargetcoverage = new File(TEST_DATA_PATH, "pcr_metrics_pertarg_new.txt");
```
Comment (akiezun) [inline src/test/java/org/broadinstitute/hellbender/tools/picard/analysis/directed/CollectTargetedPcrMetricsIntegrationTest.java]: The parameter `PER_TARGET_COVERAGE` is an output file. So you need to compare the output with the expected output (which you can get by running original picard).

File: src/test/java/org/broadinstitute/hellbender/tools/picard/analysis/directed/CollectTargetedPcrMetricsIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+public final class CollectTargetedPcrMetricsIntegrationTest extends CommandLineProgramTest {
+    private static final File TEST_DATA_PATH = new File(getTestDataDir(), "picard/analysis/directed/CollectTargetedPcrMetrics/");
+
+
+    @Test
+    public void testCollect () throws IOException {
+
+        final File input = new File(TEST_DATA_PATH, "first5Kamod_reordered.bam");
+        final File amplicon_intervals = new File(TEST_DATA_PATH, "intervals_b37_20_1.interval_list");
```
Comment (akiezun) [inline src/test/java/org/broadinstitute/hellbender/tools/picard/analysis/directed/CollectTargetedPcrMetricsIntegrationTest.java]: I'd prefer a smaller file here. Just cut the first 100 targets or so

File: src/test/java/org/broadinstitute/hellbender/tools/picard/analysis/directed/CollectTargetedPcrMetricsIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        final File target_intervals = new File(TEST_DATA_PATH, "intervals_b37_20_1.interval_list");
```
Comment (akiezun) [inline src/test/java/org/broadinstitute/hellbender/tools/picard/analysis/directed/CollectTargetedPcrMetricsIntegrationTest.java]: same here - make smaller

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,6 +1,9 @@
+
```
Comment (akiezun) [inline gradle/wrapper/gradle-wrapper.properties]: this file has  not changed - remove it from the pull request (ie replace from master)

File: src/test/java/org/broadinstitute/hellbender/tools/picard/analysis/directed/CollectTargetedPcrMetricsIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
+package org.broadinstitute.hellbender.tools.picard.analysis.directed;
+        import org.broadinstitute.hellbender.CommandLineProgramTest;
```
Comment (akiezun) [inline src/test/java/org/broadinstitute/hellbender/tools/picard/analysis/directed/CollectTargetedPcrMetricsIntegrationTest.java]: run `Reformat Code` in IntelliJ

File: src/test/java/org/broadinstitute/hellbender/tools/picard/analysis/directed/CollectTargetedPcrMetricsIntegrationTest.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        import org.broadinstitute.hellbender.utils.test.IntegrationTestSpec;
+        import org.broadinstitute.hellbender.utils.test.BaseTest;
+        import org.testng.annotations.Test;
+
+        import java.io.File;
+        import java.io.IOException;
+
+/**
```
Comment (akiezun) [inline src/test/java/org/broadinstitute/hellbender/tools/picard/analysis/directed/CollectTargetedPcrMetricsIntegrationTest.java]: remove this template comment

File: src/test/java/org/broadinstitute/hellbender/tools/picard/analysis/directed/CollectTargetedPcrMetricsIntegrationTest.java (hunk 3 above)
Comment (akiezun) [inline src/test/java/org/broadinstitute/hellbender/tools/picard/analysis/directed/CollectTargetedPcrMetricsIntegrationTest.java]: I;d prefer to use a different list for amplicons and for targets - something more similar to reality of such data.

**Comment by akiezun** [comment]
back to @dekling - looking better, small things to fix.

File: src/test/java/org/broadinstitute/hellbender/tools/picard/analysis/directed/CollectTargetedPcrMetricsIntegrationTest.java (hunk 1 above)
Comment (dekling) [inline src/test/java/org/broadinstitute/hellbender/tools/picard/analysis/directed/CollectTargetedPcrMetricsIntegrationTest.java]: Yes.  The original "pcr_metrics_pertarg_new.txt" file was generated using picard.  Do I need to compare these output files manually?

Comment (dekling) [inline src/test/java/org/broadinstitute/hellbender/tools/picard/analysis/directed/CollectTargetedPcrMetricsIntegrationTest.java]: Do I need to change this line?  
IntegrationTestSpec.assertEqualTextFiles(outfile,expectedFile, "#");  If yes, how shall I modify it?

Thanks,
-D

Comment (dekling) [inline src/test/java/org/broadinstitute/hellbender/tools/picard/analysis/directed/CollectTargetedPcrMetricsIntegrationTest.java]: @akiezun 
I added the comparison between the temp outfile and the expected but for whatever reason, Hellbender did not generate the temp. file, [pertargetcoverageout = BaseTest.createTempFile("pcr_metrics_pertarg_new_out", ".txt")].

    public void testCollect() throws IOException {

        final File input = new File(TEST_DATA_PATH, "microbam.bam");
        final File amplicon_intervals = new File(TEST_DATA_PATH, "lifted_Chr20test_targets.interval_list");
        final File target_intervals = new File(TEST_DATA_PATH, "lifted_Chr20test_regions_t.interval_list");
        final File expectedFile = new File(TEST_DATA_PATH, "PCR_new_Metrics.txt");
        final File outfile = BaseTest.createTempFile("PCRMetrics", ".txt");
        final File reference = new File(largeFileTestDir, "human_g1k_v37.20.21.fasta");
        final File pertargetcoverage = new File(TEST_DATA_PATH, "pcr_metrics_pertarg_new.txt");
        final File pertargetcoverageout = BaseTest.createTempFile("pcr_metrics_pertarg_new_out", ".txt");

        final String[] args = {
                "--INPUT", input.getAbsolutePath(),
                "--AI", amplicon_intervals.getAbsolutePath(),
                "--TI", target_intervals.getAbsolutePath(),
                "--OUTPUT", outfile.getAbsolutePath(),
                "--R", reference.getAbsolutePath(),
                "--N", "lifted_Chr20test_targets",
                "--LEVEL", "ALL_READS",
                "--PER_TARGET_COVERAGE", pertargetcoverage.getAbsolutePath(),
        };

        runCommandLine(args);
        IntegrationTestSpec.assertEqualTextFiles(outfile, expectedFile, "#");
        IntegrationTestSpec.assertEqualTextFiles(pertargetcoverageout, pertargetcoverage, "#");

Any ideas on what to do?

**Comment by dekling** [comment]
All issues addressed.  Back to @akiezun.

**Comment by akiezun** [comment]
@dekling the pull request seems broken. The changed file has no relationship to the issue. Should we close this and start over?

**Comment by dekling** [comment]
Sure.
> On Oct 19, 2015, at 2:45 PM, Adam Kiezun <notifications@github.com> wrote:
> 
> @dekling <https://github.com/dekling> the pull request seems broken. The changed file has no relationship to the issue. Should we close this and start over?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/broadinstitute/gatk/pull/996#issuecomment-149309658>.
> 

—
David E. Kling, Ph.D.
Senior Technical Writer
Broad Institute
75 Ames Street, #7003DC
Cambridge, MA 02142
O (617) 714 - 7351
C (617) 291 - 1617

**Comment by dekling** [comment]
@akiezun I believe that CollectTargetedPcrMetrics integration test... fails due to its inability to output min_normalized_coverage when using the PER_TARGET_COVERAGE option.

**Pull request closed without merging**